### ISSUE TITLE
OBIS Codes in "lesbarer" Form

### ISSUE THREAD

**Issue by Noschvie** [ixs/tasmota-sml-parser#2]
Tasmota unterstützt nun OBIS Codes in einer "lesbaren" Form.

@gemu2015 has changed the syntax to pure obis code

`1,pm(1.8.1)@1000,kWh_IN,kWh,kWh_IN,3;Wirkenergie Lieferung (+AFF)`

https://github.com/arendst/Tasmota/issues/8790#issuecomment-1407694660

**Comment by ixs** [comment]
Ohh. Nice. Danke für den Hinweis.

Das könnte man als alternatives Output Format der Meter-Definition ausgeben.
Den SML Stream muss man ja immer noch initial einmal parsen, damit man überhaupt mitbekommt, was denn der Zähler ausgibt, oder?

**Comment by Noschvie** [comment]
Wenn man die Doku vom SmartMeter hat und die gelieferten OBIS Codes kennt, ist ein Parsen des Streams eigentlich nicht mehr notwendig.

**Comment by ixs** [comment]
Richtig, aber in dem Fall braucht man das ganze Tool nicht, sondern kann sich das notwendige SML Script von Hand bauen.
Dieses Tool ist ja für den Fall, daß man halt nicht weiß, was das Smartmeter ausgibt. Oder wenn es etwas anderes als die Dokumentation ausgibt, was ja auch vorkommt.

**Comment by Noschvie** [comment]
Dieses Tool wird auch dann benötigt, wenn der Smartmeter keine OBIS Codes liefert sondern nur die jeweiligen Werte.

**Comment by hw-schrauber** [comment]
Gute Arbeit. Danke für das Tool. Habe jedoch noch ein paar kleinere Schwierigkeiten mit den Vorschlägen, die aus einem Datensatz generiert werden. Nicht alle Zeilen(Register) werden lesbar ausgegeben. Z.B. bekomme ich für das Register 96.50.1.1 (Hersteller) auf der Decoderseite "LOG" angezeigt. Bei Tasmota (UI/Console) kommt nix. Ähnlich verhält es sich mit 96.1.0.255.  Hier habe ich schon herausgefunden, dass bei Angabe von @# statt @1 ich zumindest die Ausgabe "010feA......." bekomme. Trage ich jedoch beim Hersteller statt der @1 das @# ein wird mir auch da die Server-ID angezeigt. Vielleicht habe ich es auch noch nicht richtig verstanden. In der Tasmota/SML Doku bin ich jedoch nicht fündig geworden.
Vielleicht nen Tipp, was ich tun kann?

**Comment by kongo09** [comment]
> Gute Arbeit. Danke für das Tool. Habe jedoch noch ein paar kleinere Schwierigkeiten mit den Vorschlägen, die aus einem Datensatz generiert werden. Nicht alle Zeilen(Register) werden lesbar ausgegeben. Z.B. bekomme ich für das Register 96.50.1.1 (Hersteller) auf der Decoderseite "LOG" angezeigt. Bei Tasmota (UI/Console) kommt nix. Ähnlich verhält es sich mit 96.1.0.255. Hier habe ich schon herausgefunden, dass bei Angabe von @# statt @1 ich zumindest die Ausgabe "010feA......." bekomme. Trage ich jedoch beim Hersteller statt der @1 das @# ein wird mir auch da die Server-ID angezeigt. Vielleicht habe ich es auch noch nicht richtig verstanden. In der Tasmota/SML Doku bin ich jedoch nicht fündig geworden. Vielleicht nen Tipp, was ich tun kann?

Ich bin ähnlich gescheitert und erst nach einlesen in SML habe ich händisch den Hersteller und die Zählernummer verstanden, siehe auch #6 . Leider bekomme ich das weder in Tasmota noch in diesem Tool hier wirklich dekodiert.

**Comment by c64emulator** [comment]
mit "@#x" ist es nun möglich, den Zählerstring in ASCII-Zeichen statt Hex-Werten anzuzeigen.
#https://github.com/arendst/Tasmota/discussions/16232
